controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -74,6 +74,22 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// - ocmApiUrl: Optional, defaults to 'https://api.openshift.com'
 	// +optional
 	CredentialsSecretRef *corev1.LocalObjectReference `json:"credentialsSecretRef,omitempty"`
+
+	// AWS configures aspects of the ROSA HCP workload cluster that are specific to AWS.
+	AWS AWSConfiguration `json:"aws"`
```

Comment:
why is this field required?
Given this provider can only provision AWS clusters, the naming of this field `aws` might be confusing. Do we really need to wrap the `PrivateLink` field with this?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4758#discussion_r1465343401